discovery_engine/lib/src/api/events/client_events.dart

```diff
@@ -105,6 +111,29 @@ class ClientEvent with _$ClientEvent {
     UserReaction userReaction,
   ) = UserReactionChanged;
 
+  /// Event created when the user starts a new active search.
+  @Implements<SearchClientEvent>()
+  const factory ClientEvent.searchRequested(
+    String queryTerm,
```

Comment:
should we already `assert` in this constructor that the `queryTerm` is not empty? i'd assume that empty querries don't make much sense, if they are even valid querries.

Reply:
good idea, so I've added the asserts :)